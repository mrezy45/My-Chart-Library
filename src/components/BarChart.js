import React from 'react';
import '../styles/ChartStyles.css';

const BarChart = ({ data, colors = [] }) => {
  return (
    <div className="chart-container">
      <svg width="100%" height="300px">
        {data.map((d, i) => (
          <rect
            key={i}
            x={i * 50}
            y={300 - d.value}
            width="40"
            height={d.value}
            fill={colors[i % colors.length] || 'steelblue'}
          />
        ))}
      </svg>
    </div>
  );
};

export default BarChart;