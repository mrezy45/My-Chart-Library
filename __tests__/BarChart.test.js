import React from 'react';
import { render } from '@testing-library/react';
import BarChart from '../src/components/BarChart';

describe('BarChart Component', () => {
  it('renders without crashing', () => {
    const data = [{ value: 50 }, { value: 80 }];
    const { container } = render(<BarChart data={data} colors={['#ff0000', '#00ff00']} />);
    expect(container).toMatchSnapshot();
  });

  it('renders the correct number of bars', () => {
    const data = [{ value: 50 }, { value: 80 }, { value: 30 }];
    const { container } = render(<BarChart data={data} colors={['#ff0000', '#00ff00']} />);
    
    // Find the rect elements (bars)
    const bars = container.querySelectorAll('rect');
    
    // Ensure that the number of rect elements matches the data length
    expect(bars.length).toBe(data.length);
  });
  
});
