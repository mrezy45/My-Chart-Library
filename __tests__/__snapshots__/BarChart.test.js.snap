// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BarChart Component renders without crashing 1`] = `
<div>
  <div
    class="chart-container"
  >
    <svg
      height="300px"
      width="100%"
    >
      <rect
        fill="#ff0000"
        height="50"
        width="40"
        x="0"
        y="250"
      />
      <rect
        fill="#00ff00"
        height="80"
        width="40"
        x="50"
        y="220"
      />
    </svg>
  </div>
</div>
`;
